### PR TITLE
[tests-only][full-ci] Added API tests for searching users by normal user

### PR DESCRIPTION
**Pull request by ghost**
## Description

Adding test to get user data as a normal user using search queries
```cucumber
Scenario Outline: non-admin user searches other users by e-mail
Scenario Outline: non-admin user searches for a disabled users
Scenario Outline: non-admin user searches for multiple users having same displayname
```

## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/7944

## How Has This Been Tested?

- test environment:
- test case 1:local
- test case 2:ci

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiGraphUserGroup/getUser.feature [hunk 1]
```diff
@@ -1443,3 +1443,171 @@ Feature: get users
...
+    }
+    """
+
+  @issue-7990
+  Scenario: non-admin user searches for multiple users having same displayname
+    Given the user "Admin" has created a new user with the following attributes:
+      | userName    | SameDisplayName              |
+      | displayName | Alice Hansen                 |
+      | email       | alice@example.de             |
+      | password    | containsCharacters(*:!;_+-&) |
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraphUserGroup/getUser.feature:1540]:
```suggestion
      | userName    | another-alice                |
      | displayName | Alice Hansen                 |
      | email       | another-alice@example.org    |
      | password    | containsCharacters(*:!;_+-&) |
```

File: tests/acceptance/features/apiGraphUserGroup/getUser.feature [hunk 2]
```diff
@@ -1443,3 +1443,171 @@ Feature: get users
...
+              "enum": ["Member"]
+            }
+          }
+        }
+      }
+    }
+    """
+
+  @issue-7990
+  Scenario: non-admin user searches for a disabled users
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraphUserGroup/getUser.feature:1491]: so searching disabled users is possible

CC @ScharfViktor

Comment (ScharfViktor) [reply tests/acceptance/features/apiGraphUserGroup/getUser.feature:1491]: Yes, it possible. old `sharees?search=mar` can also find disabled user
we forbid sharing to disabled user. I think the `@problem-7990` link here is incorrect

Comment (saw-jan) [reply tests/acceptance/features/apiGraphUserGroup/getUser.feature:1491]: the explanation here https://github.com/owncloud/ocis/issues/7962#issuecomment-1883179994 says that the sharing with disabled user is possible. And we added tests here https://github.com/owncloud/ocis/pull/8139

**Review by saw-jan — APPROVED** [review]
LGTM

**Review by v-scharf — APPROVED** [review]

**Pull request merged**